File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/annotation/AnnotationRequest.java
package com.portSrilanka.board_admin_backend.dto.annotation;

import com.portSrilanka.board_admin_backend.enums.AnnotationType;
import lombok.Data;

@Data
public class AnnotationRequest {
    private Long paperId;
    private Long userId;
    private AnnotationType annotationType;
    private String annotationDataJson;
    private Integer pageNumber;
}

File: board-admin-frontend/frontend/lib/features/annotations/model/annotation_request.dart
class AnnotationRequest {
  final int paperId;
  final int userId;
  final String annotationType;
  final String annotationDataJson;
  final int? pageNumber;

  AnnotationRequest({
    required this.paperId,
    required this.userId,
    required this.annotationType,
    required this.annotationDataJson,
    this.pageNumber,
  });

  Map<String, dynamic> toJson() {
    return {
      'paperId': paperId,
      'userId': userId,
      'annotationType': annotationType,
      'annotationDataJson': annotationDataJson,
      'pageNumber': pageNumber,
    };
  }
}
